packedForest/src/forestTypes/unsupervisedForests/stratifiedInNodeClassIndicesUnsupervised.h

```diff
@@ -18,22 +19,18 @@ namespace fp{
 			std::vector<int> binSamples;
 			int inSampleSize;
 			int outSampleSize;
+			double impurity;
 
 			//TODO: the following functions would benefit from Vitter's Sequential Random Sampling
 		public:
-			stratifiedInNodeClassIndicesUnsupervised(): inSamples(fpSingleton::getSingleton().returnNumClasses()), outSamples(fpSingleton::getSingleton().returnNumClasses()), inSampleSize(0), outSampleSize(0){}
+			stratifiedInNodeClassIndicesUnsupervised(): inSampleSize(0), outSampleSize(0){}
 
 
-			stratifiedInNodeClassIndicesUnsupervised(const int &numObservationsInDataSet): inSamples(fpSingleton::getSingleton().returnNumClasses()), outSamples(fpSingleton::getSingleton().returnNumClasses()), inSampleSize(0), outSampleSize(0){
-
-				createInAndOutSets(numObservationsInDataSet);
-				for(auto inSampsClass : inSamples){
-					inSampleSize += inSampsClass.size();
-				}
-
-				for(auto outSamps1 : outSamples){
-					outSampleSize += outSamps1.size();
-				}
+			stratifiedInNodeClassIndicesUnsupervised(const int &numObservationsInDataSet): inSampleSize(0), outSampleSize(0){
+				impurity = 10; //initialize to an arbitrary non zero value
```

Comment:
thanks for the comment.  i was curious about why 10.  

My understanding is this needs to be > 0 so that we process the node (unless impurity really is 0).  can you update comment with that info?